WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/management/InsightsManagementViewModel.kt

```diff
@@ -2,24 +2,115 @@ package org.wordpress.android.ui.stats.refresh.lists.sections.insights.managemen
-import org.wordpress.android.ui.pages.SnackbarMessageHolder
-import org.wordpress.android.util.map
+import org.wordpress.android.ui.stats.refresh.INSIGHTS_USE_CASE
+import org.wordpress.android.ui.stats.refresh.lists.BaseListUseCase
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type.ADDED
+import org.wordpress.android.ui.stats.refresh.lists.sections.insights.management.InsightsManagementViewModel.InsightModel.Type.REMOVED
+import org.wordpress.android.ui.stats.refresh.utils.StatsSiteProvider
 import org.wordpress.android.viewmodel.ScopedViewModel
+import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 import javax.inject.Named
 
 class InsightsManagementViewModel @Inject constructor(
-    @Named(UI_THREAD) mainDispatcher: CoroutineDispatcher
+    @Named(UI_THREAD) mainDispatcher: CoroutineDispatcher,
+    @Named(INSIGHTS_USE_CASE) val insightsUseCase: BaseListUseCase,
+    private val siteProvider: StatsSiteProvider,
+    private val statsStore: StatsStore
 ) : ScopedViewModel(mainDispatcher) {
-    private val mutableSnackbarMessage = MutableLiveData<Int>()
+    private val _removedInsights = MutableLiveData<List<InsightModel>>()
+    val removedInsights: LiveData<List<InsightModel>> = _removedInsights
 
-    val showSnackbarMessage: LiveData<SnackbarMessageHolder> = mutableSnackbarMessage.map {
-        SnackbarMessageHolder(it)
+    private val _addedInsights = MutableLiveData<List<InsightModel>>()
+    val addedInsights: LiveData<List<InsightModel>> = _addedInsights
+
+    private val _closeInsightsManagement = SingleLiveEvent<Unit>()
+    val closeInsightsManagement: LiveData<Unit> = _closeInsightsManagement
+
+    fun start() {
+        loadInsights()
     }
 
-    override fun onCleared() {
-        mutableSnackbarMessage.value = null
+    private fun loadInsights() {
+        launch {
+            val model = statsStore.getInsightsManagementModel(siteProvider.siteModel)
+            _addedInsights.value = model.addedTypes
+                    .filter { it != FOLLOWER_TOTALS && it != ANNUAL_SITE_STATS }
+                    .map { InsightModel(it, ADDED) }
+
+            _removedInsights.value = model.removedTypes
+                    .filter { it != FOLLOWER_TOTALS && it != ANNUAL_SITE_STATS }
+                    .map { InsightModel(it, REMOVED) }
+        }
+    }
+
+    fun onSaveInsights() {
+        launch {
+            val addedTypes = addedInsights.value?.map { it.insightsTypes } ?: emptyList()
+            val removedTypes = removedInsights.value?.map { it.insightsTypes } ?: emptyList()
+            val model = InsightTypesModel(addedTypes, removedTypes + FOLLOWER_TOTALS + ANNUAL_SITE_STATS)
+            statsStore.updateTypes(siteProvider.siteModel, model)
+
+            launch {
```

Comment:
NIT: if we switch the `launch` and the `refreshData` call, we can remove the `launch` wrapper :-)